### PR TITLE
support reading and accessing external_id into profile

### PR DESCRIPTION
**Pull request by yhager**

### CODE REVIEW

File: aws-cpp-sdk-core-tests/aws/config/AWSProfileConfigLoaderTest.cpp [hunk 1]
```diff
@@ -36,6 +36,7 @@ static void WriteDefaultConfigFile(Aws::OStream& stream, bool useProfilePrefix =
 
     stream << "[ " << profilePrefix << "assumes_role]" << std::endl;
     stream << "role_arn =arn:aws:iam::123456789:role/foo" << std::endl;
+    stream << "external_id = +CsOo4kcN4UKViieKhF7qDs46qmM974XLOzp+TEs/9M=" << std::endl;
```
Comment (marcomagdy) [inline aws-cpp-sdk-core-tests/aws/config/AWSProfileConfigLoaderTest.cpp]: I hope this is not a real key.
Please replace with some other text just in case.

Comment (yhager) [reply aws-cpp-sdk-core-tests/aws/config/AWSProfileConfigLoaderTest.cpp]: no, it's not a real key. This was generated using `openssl rand -base64 32`. I'll replace it with some other text, just for the looks of it.

**Comment by marcomagdy** [comment]
Merged internally, will run through our CI and push it out by EOD tomorrow.
Thank you!

**Comment by marcomagdy** [comment]
Merged in v1.6.6

**Pull request closed without merging**